browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/data/UserDashboardViewModel.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.okta.android.samples.browser_sign_in.data
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.okta.android.samples.browser_sign_in.BuildConfig
+import com.okta.authfoundation.claims.email
+import com.okta.authfoundation.claims.name
+import com.okta.authfoundation.client.OidcClientResult
+import com.okta.authfoundationbootstrap.CredentialBootstrap
+import com.okta.webauthenticationui.WebAuthenticationClient.Companion.createWebAuthenticationClient
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import timber.log.Timber
+
+class UserDashboardViewModel : ViewModel() {
+    private val _loginState = MutableLiveData<LoginState>(LoginState.LoggedIn)
+    private val _userGreetingState = MutableLiveData<UserGreetingState>(UserGreetingState.Loading)
+    private val _userInfoState = MutableLiveData<UserInfoState>(UserInfoState.Empty)
+
+    val loginState: LiveData<LoginState> = _loginState
+    val userGreetingState: LiveData<UserGreetingState> = _userGreetingState
+    val userInfoState: LiveData<UserInfoState> = _userInfoState
+
+    init {
+        viewModelScope.launch {
+            refreshCurrentLoginState()
+        }
+    }
+
+    fun logoutOfBrowser(context: Context) {
+        viewModelScope.launch {
+            _userGreetingState.value = UserGreetingState.Loading
```

Comment:
This state seems like it should be unrelated to logout. It might be best to use a new state for logout only. (Idle, Loading, LoggedOut) or similar.